src/CoolFileSystem.cpp

```diff
@@ -994,91 +710,174 @@ bool CoolFileSystem::fileUpdate(String update, const char *path) {
   return (true);
 }
 
+
 /**
- *  CoolFileSystem::incrementsavedData():
- *  This method is provided to increment the
- *  savedData flag
+ *  CoolFileSystem::saveMessageToFile(():
+ *  This method is provided to get the
+ *  savedData flag from the file system
  *
- *  \return true if successful , false otherwise
  *
  */
-bool CoolFileSystem::incrementsavedData() {
-
+bool CoolFileSystem::saveMessageToFile(const char *data) {
+  String lastName = "0";
 #if DEBUG == 1
+  FSInfo fs_info;
 
-  Serial.println(F("Entering CoolFileSystem.incrementsavedData()"));
-
+  if (SPIFFS.info(fs_info) == true) {
+    Serial.print(F("used bytes/total bytes : "));
+    Serial.print(fs_info.usedBytes);
+    Serial.print(F("/"));
+    Serial.print(fs_info.totalBytes);
+    Serial.println();
+  }
 #endif
-  // open file
-  File file = SPIFFS.open("/savedDataFlag.txt", "w");
-  // read file
-  if (!file) {
-    Serial.println(F("failed to open savedDataFlag.txt"));
-
-    return (false);
-  } else {
-    // write to file
-    file.print(this->savedData);
 
-    file.print(" ");
+  int nextLog = lastFileSaved();
+  nextLog++;
+  char nextName[32] = "0";
+  snprintf(nextName, 32, "/log/%ld.json", nextLog);
 
-    file.println(this->linesToSkip);
+#if DEBUG == 1
+  Serial.println(F("number for next filename : "));
+  Serial.println(nextLog);
+  Serial.print(F("Name for next file : "));
+  Serial.println(nextName);
+#endif
 
-    file.close();
+  if (!SPIFFS.exists(nextName))
+  {
+    File f = SPIFFS.open(nextName, "w");
+    f.print(data);
+    f.close();
+    Serial.print(F("File Saved as : "));
+    Serial.println(nextName);
+  } else return false;
 
 #if DEBUG == 1
+  if (SPIFFS.info(fs_info) == true) {
+    Serial.print(F("used bytes/total bytes : "));
+    Serial.print(fs_info.usedBytes);
+    Serial.print(F("/"));
+    Serial.print(fs_info.totalBytes);
+    Serial.println();
+  }
+#endif
 
-    Serial.print(F("number of lines to read :"));
-    Serial.println(this->savedData);
-
-    Serial.print(F("number of lines to skip :"));
-    Serial.println(this->linesToSkip);
+  return true;
+}
 
-#endif
 
-    return (true);
+/**
+ *  CoolFileSystem::getsavedData():
+ *  This method is provided to get the
+ *  savedData flag from the file system
+ *
+ *
+ */
+bool CoolFileSystem::isFileSaved() {
+  Dir dir = SPIFFS.openDir("/log");
+  if (dir.next()) {
+    return true;
   }
+  else return false;
 }
+
+
 /**
  *  CoolFileSystem::getsavedData():
  *  This method is provided to get the
  *  savedData flag from the file system
  *
  *
  */
-void CoolFileSystem::getsavedData() {
+int CoolFileSystem::lastFileSaved() {
+  String lastName = "0";
```

Comment:
I think `index` and `lastName` variables are not really useful here. You can live with just `next` and `temp`.